src/framework/model.js

```diff
@@ -0,0 +1,211 @@
+            return;
+        this._mainColor = value;
+        this.notify('main-color');
+    }
+
+    get accent_color() {
+        return this._accentColor;
+    }
+
+    set accent_color(value) {
+        if ('_accentColor' in this && this._accentColor === value)
+            return;
+        this._accentColor = value;
+        this.notify('accent-color');
+    }
+
+    get info_color() {
+        return this._infoColor;
+    }
+
+    set info_color(value) {
+        if ('_infoColor' in this && this._infoColor === value)
+            return;
+        this._infoColor = value;
+        this.notify('info-color');
+    }
+
+    get font() {
+        return this._font;
+    }
+
+    set font(value) {
+        if ('_font' in this && this._font === value)
+            return;
+        this._font = value;
+        this.notify('font');
+    }
+
+    get font_size() {
+        return this._fontSize;
+    }
+
+    set font_size(value) {
+        if ('_fontSize' in this && this._fontSize === value)
+            return;
+        this._fontSize = value;
+        this.notify('font-size');
+    }
+
+    get card_borders() {
+        return this._cardBorders;
+    }
+
+    set card_borders(value) {
+        if ('_cardBorders' in this && this._cardBorders === value)
+            return;
+        this._cardBorders = value;
+        this.notify('card-borders');
+    }
+
+    get border_color() {
+        return this._borderColor;
+    }
+
+    set border_color(value) {
+        if ('_borderColor' in this && this._borderColor === value)
+            return;
+        this._borderColor = value;
+        this.notify('border-color');
+    }
+
+    get text() {
+        return this._text;
+    }
+
+    set text(value) {
+        if ('_text' in this && this._text === value)
+            return;
+        this._text = value;
+        this.notify('text');
+    }
+
+    get order() {
+        return this._order;
+    }
+
+    set order(value) {
+        if ('_order' in this && this._order === value)
+            return;
+        this._order = value;
+        this.notify('order');
+    }
+
+    get layout() {
+        return this._layout;
+    }
+
+    set layout(value) {
+        if ('_layout' in this && this._layout === value)
+            return;
+        this._layout = value;
+        this.notify('layout');
+    }
+
+    get filter() {
+        return this._filter;
+    }
+
+    set filter(value) {
+        if ('_filter' in this && this._filter === value)
+            return;
+        this._filter = value;
+        this.notify('filter');
+    }
+
+    get hover_sound() {
+        return this._hoverSound;
+    }
+
+    set hover_sound(value) {
+        if ('_hoverSound' in this && this._hoverSound === value)
+            return;
+        this._hoverSound = value;
+        this.notify('hover-sound');
+
+        if (value !== 'none') {
+            this._clickSound = 'none';
+            this.notify('click-sound');
```

Comment:
No fundamental reason, but this made it simpler to implement and I also thought it would make it difficult to play a melody on the piano, for example, if you were also getting hover sounds in between. I can remove that restriction, but if so we should create another issue for it.

Reply:
Sounds good